lib/pbench/agent/tool_meister.py

```diff
@@ -1289,45 +1289,45 @@ def _create_tar(
         self,
         directory: Path,
         tar_file: Path,
-        tar_args: List = [],
         retry: bool = False,
     ) -> subprocess.CompletedProcess:
         """
         Creates a tar file at a given tar file path.
-        If an error occurs it will retry by suppressing all the warnings.
+        If an error occurs, it will retry with all warnings suppressed.
         """
-        if not tar_args:
-            tar_args = [
-                self.tar_path,
-                "--create",
-                "--xz",
-                "--force-local",
-                f"--file={tar_file}",
-                directory.name,
-            ]
+
+        def tar(directory: Path, tar_args: List):
+            return subprocess.run(
+                tar_args,
+                cwd=directory.parent,
+                stdin=None,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+            )
 
         # Invoke tar directly for efficiency.
-        cp = subprocess.run(
-            tar_args,
-            cwd=directory.parent,
-            stdin=None,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
-        )
+        tar_args = [
+            self.tar_path,
+            "--create",
+            "--xz",
+            "--force-local",
+            f"--file={tar_file}",
+            directory.name,
+        ]
+
+        cp = tar(directory, tar_args)
         if cp.returncode != 0:
             self.logger.warning(
-                "Tar ball creation failed with '%s', %s; returncode: %d",
+                "Tar ball creation failed with '%s' and returncode: %d, on directory %s; %s",
                 cp.stdout.decode("utf-8"),
-                "re-trying ignoring all warnings using --warning=none" if retry else "",
                 cp.returncode,
+                directory,
+                "re-trying ignoring all warnings using --warning=none" if retry else "",
```

Comment:
You should have included the trailing "; " as well, because right now the non-`retry` case will look weird. However, I think it'd be better to just drop this entire phrase anyway. If you want, just include the value of the `retry` parameter.

Reply:
For non-retry cases it looks like this:
`
__logger__ _create_tar 1320 -- Tar ball creation failed with 'Some error running tar' and returncode: 1, on directory /tmp/pytest-of-nikhilpalaskar/pytest-193/agent-tmp0; 
`

and for the retry case it looks like this 
`__logger__ _create_tar 1320 -- Tar ball creation failed with 'Some error running tar' and returncode: 1, on directory /tmp/pytest-of-nikhilpalaskar/pytest-192/agent-tmp0; re-trying ignoring all warnings using --warning=none` 

I dont think hey look weird, its just verbose warning message

Reply:
What looks weird is the trailing semicolon at the end of the message.  Messages normally end with a period (or exclamation mark, or question mark...), if they end with any punctuation at all; in English text, a semicolon implies that there is another independent clause coming....